syntax = "proto3";

service YoloService {
    rpc DetectStream (stream DetectRequest) returns (stream DetectResponse) {}
    rpc Detect (DetectRequest) returns (DetectResponse) {}
}

message DetectRequest {
    bytes image_data = 1; // Encoded image data
}

message DetectResponse {
    string results = 1;
}

service Llama2Service {
    rpc ChatRequest (PromptRequest) returns (PromptResponse) {}
}

message PromptRequest {
    string json_data = 1; // prompt
    optional bytes image_data = 2; // Encoded image data
}

message PromptResponse {
    string json_data = 1; // response
}

service LlavaService {
    rpc VisionRequest (PromptRequest) returns (PromptResponse) {}
}
